system/jlib/jptree.hpp

```diff
@@ -313,6 +313,22 @@ inline static bool isValidXPathChr(char c)
     return ('\0' != c && (isalnum(c) || strchr(validChrs, c)));
 }
 
+static const char *validXPathValueChrs = "!#$%^&()+,-.:;?@_{|}~"; //Do I miss any?
+inline static bool isValidXPathValueChr(char c)
+{
+    return ('\0' != c && (isalnum(c) || strchr(validXPathValueChrs, c)));
+}
+inline static bool isValidXPathValue(const char *v)
+{
+    const char *p = v;
```

Comment:
I don't think you actually need p.  The following should work:
while (*v)
{
     if (!isValidXPathValueChr(*v++)) return false;
}

Reply:
Yes, I do not need p.
The do...while is used to check an empty v.